python/scenario/judge_agent.py

```
@@ -688,91 +1258,66 @@ async def call(
-        # relaxes this pin to "required" so the judge can use discovery tools,
-        # but _run_discovery_loop forces a final verdict if a required judgment
-        # answers continue_test there, so the terminal contract holds on both
-        # paths.
+        exhausted_entry = bool(discovery_recap)
+        if not exhausted_entry:
+            if is_large_span_trace:
+                tools = self._build_progressive_discovery_tools() + tools
+            if view.is_large_transcript:
+                tools = self._build_transcript_discovery_tools() + tools
+        if wait_extension_available:
+            tools = [_build_wait_for_traces_tool()] + tools
```

Ruff (0.16.1)

[warning] Consider `[_build_wait_for_traces_tool(), *tools]` instead of concatenation

Replace with `[_build_wait_for_traces_tool(), *tools]`

(RUF005)